tests/worker/test_workflow.py

```diff
@@ -3213,21 +3216,57 @@ async def test_manual_result_type(client: Client):
         assert res4 == ManualResultType(some_string="from-query")
 
 
+@activity.defn
+async def wait_forever_activity() -> None:
+    await asyncio.Future()
+
+
+@workflow.defn
+class WaitForeverWorkflow:
+    @workflow.run
+    async def run(self) -> None:
+        await asyncio.Future()
+
+
 @workflow.defn
-class SwallowGeneratorExitWorkflow:
+class CacheEvictionTearDownWorkflow:
     def __init__(self) -> None:
         self._signal_count = 0
 
     @workflow.run
     async def run(self) -> None:
+        # Start several things in background. This is just to show that eviction
+        # can work even with these things running.
+        tasks = [
+            asyncio.create_task(
+                workflow.execute_activity(
+                    wait_forever_activity, start_to_close_timeout=timedelta(hours=1)
+                )
+            ),
+            asyncio.create_task(
+                workflow.execute_child_workflow(WaitForeverWorkflow.run)
+            ),
+            asyncio.create_task(asyncio.sleep(1000)),
+            asyncio.shield(
+                workflow.execute_activity(
+                    wait_forever_activity, start_to_close_timeout=timedelta(hours=1)
+                )
+            ),
+        ]
+        gather_fut = asyncio.gather(*tasks, return_exceptions=True)
         try:
             # Wait for signal count to reach 2
+            await asyncio.sleep(0.01)
```

Comment:
What's this guy here for?

Reply:
Hrmm, I originally thought I added it just to have more things to try to trip up eviction, but I am seeing some oddities if I remove it. I am investigating.

Reply:
After testing, I am seeing some strange server/core behavior here. If you remove this and the other `0.01` sleep in this test, it hangs. If you enable logging of the protos (i.e. setting `LOG_PROTOS = True` in `worker/_workflow.py`, it hangs but for 10s. There is some timing issue causing hang somewhere (and even with logging, the 10s hang is confusing too).

If you have the time, can you help debug this from a core POV? To replicate, clone and run things for local env (https://github.com/temporalio/sdk-python?tab=readme-ov-file#local-sdk-development-environment) and then run `poe test -s --log-cli-level=DEBUG -k test_cache_eviction_tear_down`. Then remove the two 0.01 sleeps and run again and see if it hangs for you. Similarly, enable that logging and see if it just has the 10s hiccup.

Reply:
Ok, I cannot replicate against a modern server, only against an old CLI server. 

Overall, yes, this `asyncio.sleep(0.01)` is just there for trying to add extra things to trip up the task collector, but can be removed and work without issue.